src/pages-and-resources/discussions/app-config-form/AppConfigForm.jsx

```diff
@@ -50,9 +59,16 @@ function AppConfigForm({
 
   // This is a callback that gets called after the form has been submitted successfully.
   const handleSubmit = useCallback((values) => {
-    // Note that when this action succeeds, we redirect to pagesAndResurcesPath in the thunk.
-    dispatch(saveAppConfig(courseId, selectedAppId, values, pagesAndResourcesPath));
-  }, [courseId, selectedAppId, courseId]);
+    // FIXME: this should check whether form values have changed as well!
```

Comment:
Is there some way I can check this? I can't find anything in Formik that would help me here, so the only solution I see is explicitly calling a function to check deep equality.

Reply:
@kewne I am confused why we need to check deep equality here. can you please explain? so that I can look for a solution from that perspective.

Reply:
@mehaknasir I'm assuming changing the configuration values (e.g. `consumerKey` and `consumerSecret`) could cause the posts to be "lost" as well...I'm admittedly not too familiar with how the integration works.

Reply:
Okay, got it I will get back to your original question shortly, I am also short on context here. But let me dig deeper into it.

Reply:
so about your use case, we have the old configurations saved in redux and new ones in formik.  also the initial values we pass in formik can help. in that way you will have the two object which you can deep compare by using some [manual](https://dmitripavlutin.com/how-to-compare-objects-in-javascript/) method or using [lodash](https://stackoverflow.com/questions/31683075/how-to-do-a-deep-comparison-between-2-objects-with-lodash) . Hope it helps you create a solution for the issue you are facing.

Reply:
👍 I'll go with lodash then.

As a note, related to my previous comment about the structure for this, [Formik does expose a `dirty` prop](https://formik.org/docs/api/formik#dirty-boolean) which does this, it just can't be reached from this part of the code.

Reply:
@mehaknasir Done, I've tested the change and it seems to work properly.